studio/state/table-editor.tsx

```diff
@@ -0,0 +1,30 @@
+import { PropsWithChildren, createContext, useContext, useRef } from 'react'
```

Comment:
this file should be `.ts`?

Reply:
Nope, it has some JSX in it! (`TableEditorStateContextProvider`)